src/operations/search_indexes/list.ts

```diff
@@ -0,0 +1,22 @@
+import type { Document } from 'bson';
+
+import type { Collection } from '../../collection';
+import { AggregationCursor } from '../../cursor/aggregation_cursor';
+import type { AggregateOptions } from '../aggregate';
+
+/** @public */
+export type ListSearchIndexesOptions = AggregateOptions;
+
+/** @public */
+export class ListSearchIndexesCursor extends AggregationCursor<{ name: string }> {
+  /** @internal */
+  static create<T extends Document>(
```

Comment:
Preference, mostly.  It's effectively the same, but I prefer factory functions over constructor subclassing because they can abstract details of object's construction and move logic out of the constructor.

Would you prefer I move the logic to ListSearchIndexesCursor.constructor?

Reply:
It is effectively the same, so is it only different for preference reasons? I'm not against your preference, I just don't want code that's unique to one implementation without providing meaning to the next engineer who comes along

Also can we move this to our cursors directory?